src/main/java/org/apache/mesos/curator/CuratorStateStore.java

```diff
@@ -137,18 +138,20 @@ public void clearFrameworkId() throws StateStoreException {
     }
 
     @Override
-    public Protos.FrameworkID fetchFrameworkId() throws StateStoreException {
+    public Optional<Protos.FrameworkID> fetchFrameworkId() throws StateStoreException {
         try {
             logger.debug("Fetching FrameworkID from '{}'", fwkIdPath);
             byte[] bytes = curator.fetch(fwkIdPath);
             if (bytes.length > 0) {
-                return Protos.FrameworkID.parseFrom(bytes);
+                return Optional.of(Protos.FrameworkID.parseFrom(bytes));
             } else {
                 throw new StateStoreException(String.format(
                         "Failed to retrieve FrameworkID in '%s'", fwkIdPath));
             }
+        } catch (KeeperException.NoNodeException e) {
+            logger.warn("No FrameworkId found.");
+            return Optional.empty();
```

Comment:
It distinguishes between two cases.  The thrown exception indicates that something bad and unexpected happened.  The Optional.empty() indicates that something hasn't been set before which is an expected scenario.  Clients should know how to handle the the expected empty case as well as unexpected communication errors.

This also helps the fact that our logs were filled with stack traces in expected situations which was confusing users.